### ISSUE TITLE
Some issue of VSTS url while using PAT

### ISSUE THREAD

**Issue by roger199585** [semantic-release/semantic-release#741]
## Current behavior
- Wrong repository while running `verifyAuth`
- `verifyAuth` check the verification by `git push --dry-run https://companyname.visualstudio.com/DefaultCollection/MyRepo.git HEAD:BUILD`
- The log is below
- The issue is in `get-git-auth-url.js`

## Expected behavior
On VSTS the <repositoryUrl> we need is like `https://companyname.visualstudio.com/DefaultCollection/_git/MyRepo`,
I have try it and it work when I change the `<repositoryUrl>` into the format above.
I'll send the PR later.

## Environment

- **semantic-release** version: 15.1.5
- CI environment: VSTS
- CI logs: 
```
2018-04-10T12:00:43.9592602Z fatal: Authentication failed for 'https://[secure]@joueapp.visualstudio.com/Joue/NgxSidebar.git/'
2018-04-10T12:00:43.9618081Z 
2018-04-10T12:00:43.9632042Z 
2018-04-10T12:00:43.9660310Z     at makeError (/vsts/agent/_work/1/s/node_modules/semantic-release/node_modules/execa/index.js:172:9)
2018-04-10T12:00:43.9691120Z     at Promise.all.then.arr (/vsts/agent/_work/1/s/node_modules/semantic-release/node_modules/execa/index.js:277:16)
2018-04-10T12:00:43.9719535Z     at <anonymous>
2018-04-10T12:00:43.9749730Z     at process._tickCallback (internal/process/next_tick.js:188:7)
2018-04-10T12:00:43.9777288Z [Semantic release]: EGITNOPERMISSION The push permission to the Git repository is required.
2018-04-10T12:00:43.9805753Z semantic-release cannot push the version tag to the branch BUILD on remote Git repository.
2018-04-10T12:00:43.9820889Z 
2018-04-10T12:00:43.9849821Z Please refer to the authentication configuration documentation (https://github.com/semantic-release/semantic-release/blob/caribou/docs/usage/ci-configuration.md#authentication) to configure the Git credentials on your CI environment.
2018-04-10T12:00:43.9861303Z 
2018-04-10T12:00:43.9869573Z 
2018-04-10T12:00:43.9886799Z { SemanticReleaseError: The push permission to the Git repository is required.
2018-04-10T12:00:43.9903773Z     at module.exports (/vsts/agent/_work/1/s/node_modules/semantic-release/lib/get-error.js:6:10)
2018-04-10T12:00:43.9922450Z     at run (/vsts/agent/_work/1/s/node_modules/semantic-release/index.js:51:11)
2018-04-10T12:00:43.9940073Z     at <anonymous>
2018-04-10T12:00:43.9956623Z     at process._tickCallback (internal/process/next_tick.js:188:7)
2018-04-10T12:00:43.9973157Z   name: 'SemanticReleaseError',
2018-04-10T12:00:43.9990080Z   code: 'EGITNOPERMISSION',
2018-04-10T12:00:44.0008210Z   details: '**semantic-release** cannot push the version tag to the branch `BUILD` on remote Git repository.\n\nPlease refer to the [authentication configuration documentation](https://github.com/semantic-release/semantic-release/blob/caribou/docs/usage/ci-configuration.md#authentication) to configure the Git credentials on your CI environment.',
2018-04-10T12:00:44.0027178Z   semanticRelease: true }
2018-04-10T12:00:44.0129602Z ##[error]Bash exited with code '1'.
```

**Comment by roger199585** [comment]
In `/lib/get-git-auth-url.js` line 51.
```
return gitCredentials
      ? format({...parse(`${gitUrlParse(repositoryUrl).toString(protocol)}.git`), ...{auth: gitCredentials}})
      : repositoryUrl;
```
The `repositoryUrl` we get from current version of `semantic-release` is like the `URL` below
`https://[secure]@conpanyname.visualstudio.com/DefaultCollection/MyRepo.git`.
However, I think the `RepositoryUrl` that VSTS expected is `https://[secure]@conpanyname.visualstudio.com/DefaultCollection/_git/MyRepo`

**Comment by pvdlg** [comment]
In `https://[secure]@conpanyname.visualstudio.com/DefaultCollection/_git/MyRepo` is the `_git` before `MyRepo`necessary?
It's not really if that is supposed to be there or not from your example.

What do you get if you do `git push --dry-run https://<TOKEN>@companyname.visualstudio.com/DefaultCollection/MyRepo HEAD:BUILD` (without `.git`)? I understand you are not asked the password but what is the return code of the command?

**Comment by roger199585** [comment]
This is the error log while running with `.git`
```
semantic-release:git Error: Command failed: git push --dry-run https://name.visualstudio.com/Collection/_git/Repo HEAD:BUILD
  semantic-release:git fatal: Authentication failed for 'https://name.visualstudio.com/Collection/_git/Repo/'
  semantic-release:git 
  semantic-release:git 
  semantic-release:git     at makeError (/transcode/node_modules/semantic-release/node_modules/execa/index.js:172:9)
  semantic-release:git     at Promise.all.then.arr (/transcode/node_modules/semantic-release/node_modules/execa/index.js:277:16)
  semantic-release:git     at <anonymous>
  semantic-release:git     at process._tickCallback (internal/process/next_tick.js:188:7) +0ms

  semantic-release:git Error: Command failed: git push --dry-run https://[secure]@name.visualstudio.com/Collection/Repo.git HEAD:BUILD
  semantic-release:git fatal: Authentication failed for 'https://[secure]@name.visualstudio.com/Collection/Repo.git/'
  semantic-release:git 
  semantic-release:git 
  semantic-release:git     at makeError (/transcode/node_modules/semantic-release/node_modules/execa/index.js:172:9)
  semantic-release:git     at Promise.all.then.arr (/transcode/node_modules/semantic-release/node_modules/execa/index.js:277:16)
  semantic-release:git     at <anonymous>
  semantic-release:git     at process._tickCallback (internal/process/next_tick.js:188:7) +1s
[Semantic release]: EGITNOPERMISSION The push permission to the Git repository is required.
semantic-release cannot push the version tag to the branch BUILD on remote Git repository.
```

And this is the error the is running without `.git`
```
semantic-release:git Error: Command failed: git push --dry-run https://name.visualstudio.com/Collection/_git/Repo HEAD:BUILD
  semantic-release:git fatal: Authentication failed for 'https://name.visualstudio.com/Collection/_git/Reop/'
  semantic-release:git 
  semantic-release:git 
  semantic-release:git     at makeError (/transcode/node_modules/semantic-release/node_modules/execa/index.js:172:9)
  semantic-release:git     at Promise.all.then.arr (/transcode/node_modules/semantic-release/node_modules/execa/index.js:277:16)
  semantic-release:git     at <anonymous>
  semantic-release:git     at process._tickCallback (internal/process/next_tick.js:188:7) +0ms
https://joueapp.visualstudio.com/Joue/_git/NgxSidebar
URL: https://[secure]@joueapp.visualstudio.com/Joue/NgxSidebar.git
  semantic-release:git Error: Command failed: git push --dry-run https://[secure]@name.visualstudio.com/Collection/Repo HEAD:BUILD
  semantic-release:git fatal: Authentication failed for 'https://[secure]@name.visualstudio.com/Collection/Repo/'
  semantic-release:git 
  semantic-release:git 
  semantic-release:git     at makeError (/transcode/node_modules/semantic-release/node_modules/execa/index.js:172:9)
  semantic-release:git     at Promise.all.then.arr (/transcode/node_modules/semantic-release/node_modules/execa/index.js:277:16)
  semantic-release:git     at <anonymous>
  semantic-release:git     at process._tickCallback (internal/process/next_tick.js:188:7) +1s
[Semantic release]: EGITNOPERMISSION The push permission to the Git repository is required.
semantic-release cannot push the version tag to the branch BUILD on remote Git repository.
```

No matter we have `.git` in the `URL` or not. They both `Authentication failed`, so I think `/_git/` before `Repo` might be necessary. But I'm not really sure if the error just happened on my machine.
The format that work on my machine is `https://[secure]@conpanyname.visualstudio.com/DefaultCollection/_git/MyRepo`.

**Comment by pvdlg** [comment]
Regarding the `.git` the proper fix is to fix IonicaBizau/git-url-parse#69, `update git-url-parse`, then remove the `.git` concatenated by semantic-release. This way the `.git` will be added at the end of the converted URL only if it was present in the original URL.

Regarding the `_git` it seems you already opened IonicaBizau/git-url-parse#67.
